src/shared-api-objects/current-user.ts

```diff
@@ -33,8 +33,11 @@ export function refetchCurrentUser() {
   userSubject.next(openmrsObservableFetch("/ws/rest/v1/session"));
 }
 
-export function userHasAccess(requiredPrivilege, user) {
-  return user.privileges.find(p => requiredPrivilege === p.display);
+export function userHasAccess(requiredPrivilegeOrRole, user) {
+  return (
+    user.privileges.find(p => requiredPrivilegeOrRole === p.display) ||
+    user.roles.find(r => requiredPrivilegeOrRole === r.display)
```

Comment:
Do you know if there is ever an overlap between the names for privileges and roles? if so, this addition to the function would make it impossible to test for either permission-specific or role-specific access. If not, then maybe we don't worry about it.

Reply:
Presently, we don't have a policy on how to do permissions on functionality. I don't believe we should be using "roles" to figure out if access is appropriate: roles may changed or be added; if we make this role based, we will have to update the code as roles are changed. 

I think we should restrict this to the privilege level. 

But what privilege? That I don't know. Currently OpenMRS has a number of built in privileges. We could use these or develop a MF specific set of privileges to be used for allowing access to various functionality. This is a larger discussion. However, I don't think we need an answer to this to move forward on this ticket.